src/ValueHandler/ChannelPricingValueHandler.php

```diff
@@ -85,7 +107,19 @@ public function handle($subject, string $attribute, array $value): void
                     $channelPricing = $this->channelPricingFactory->createNew();
                     $channelPricing->setChannelCode($channel->getCode());
                 }
-                $channelPricing->setPrice((int) round($price * 100));
+
+                if (!$this->propertyAccessor->isWritable($channelPricing, $this->syliusPropertyPath)) {
+                    throw new \RuntimeException(
+                        sprintf(
+                            'Property path "%s" is not writable on %s.',
+                            $this->syliusPropertyPath,
+                            get_class($channelPricing)
+                        )
+                    );
+                }
+
+                $this->propertyAccessor->setValue($channelPricing, $this->syliusPropertyPath, (int) round($price * 100));
+                Assert::isInstanceOf($channelPricing, ChannelPricingInterface::class);
```

Comment:
```suggestion

                if ($this->propertyAccessor === null) {
                    $channelPricing->setPrice((int) round($price * 100));
                } else {
                    $this->propertyAccessor->setValue($channelPricing, $this->syliusPropertyPath, (int) round($price * 100));
                    Assert::isInstanceOf($channelPricing, ChannelPricingInterface::class);
                }                    
```

Reply:
Why remove the RuntimeException if the property is not writable?

Reply:
There's no need: it will be the property accessor itself to throw an exception in case the property is not writeable. Less code is better.